tests/pagination/tests.py

```diff
@@ -318,12 +319,16 @@ def test_page_getitem(self):
         self.assertIsInstance(p.object_list, list)
 
     def test_paginating_unordered_queryset_raises_warning(self):
-        msg = (
+        with warnings.catch_warnings(record=True) as warns:
+            warnings.filterwarnings('always', category=UnorderedObjectListWarning)
+            Paginator(Article.objects.all(), 5)
+        self.assertEqual(len(warns), 1)
+        warning = warns[0]
+        self.assertEqual(str(warning.message), (
             "Pagination may yield inconsistent results with an unordered "
             "object_list: <QuerySet [<Article: Article 1>, "
             "<Article: Article 2>, <Article: Article 3>, <Article: Article 4>, "
             "<Article: Article 5>, <Article: Article 6>, <Article: Article 7>, "
             "<Article: Article 8>, <Article: Article 9>]>"
-        )
-        with self.assertRaisesMessage(UnorderedObjectListWarning, msg):
-            Paginator(Article.objects.all(), 5)
+        ))
+        self.assertEqual(warning.filename, __file__)
```

Comment:
Looks good but if you think you have to explain it, maybe a comment to the effect of ``# The warning points at the Paginator caller (i.e. the stacklevel is appropriate).`` would be appropriate.